submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/db/converters/from/DocSectionConverterTest.kt

```diff
@@ -0,0 +1,170 @@
+    }
+
+    private fun assertThatSections(result: DocSection) {
+        val section1 = result.sections[0] as Either.Left<DocSection>
+        assertThat(section1.a.accNo).isEqualTo(AccNo2)
+        assertThat(section1.a.type).isEqualTo(Type2)
+        assertThat(section1.a.fileList).isEqualTo(docFileList2)
+        assertThat(section1.a.attributes).isEqualTo(listOf(docAttribute2))
+
+        val section2 = result.sections[1] as Either.Right<DocSectionTable>
+        assertThat(section2.b.sections[0].accNo).isEqualTo(AccNo3)
+        assertThat(section2.b.sections[0].type).isEqualTo(Type3)
+        assertThat(section2.b.sections[0].attributes).isEqualTo(listOf(docAttribute3))
+    }
+
+    private fun createDocSectionDocument(): Document {
+        val sectionDoc = Document()
+        sectionDoc[classField] = docSectionClass
+        sectionDoc[DocSectionConverter.secAccNo] = AccNo1
+        sectionDoc[DocSectionConverter.secType] = Type1
+        sectionDoc[DocSectionConverter.secAttributes] = listOf(attributeDocument1)
+        sectionDoc[DocSectionConverter.secFileList] = fileListDocument1
+        sectionDoc[DocSectionConverter.secSections] = listOf(
+            createInternalDocSection(AccNo2, Type2, attributeDocument2, fileListDocument2),
+            createDocSectionTable(AccNo3, Type3, attributeDocument3)
+        )
+        sectionDoc[DocSectionConverter.secFiles] = listOf(fileDocument1, fileTableDocument1)
+        sectionDoc[DocSectionConverter.secLinks] = listOf(linkDocument1, linkTableDocument1)
+        return sectionDoc
+    }
+
+    private fun basicDocSection(accNo: String, type: String, attributeDocument: Document): Document {
+        val sectionDoc = Document()
+        sectionDoc[classField] = docSectionClass
+        sectionDoc[DocSectionConverter.secAccNo] = accNo
+        sectionDoc[DocSectionConverter.secType] = type
+        sectionDoc[DocSectionConverter.secAttributes] = listOf(attributeDocument)
+        return sectionDoc
+    }
+
+    private fun createInternalDocSection(accNo: String, type: String, attributeDocument: Document, fileListDocument: Document): Document {
+        val sectionDoc = basicDocSection(accNo, type, attributeDocument)
+        sectionDoc[DocSectionConverter.secFileList] = fileListDocument
+        sectionDoc[DocSectionConverter.secSections] = listOf<Document>()
+        sectionDoc[DocSectionConverter.secFiles] = listOf<Document>()
+        sectionDoc[DocSectionConverter.secLinks] = listOf<Document>()
+        return sectionDoc
+    }
+
+    private fun createDocSectionTable(accNo: String, type: String, attributeDocument: Document): Document {
+        val sectionTableDocument = Document()
+        sectionTableDocument[classField] = docSectionTableClass
+        sectionTableDocument[DocSectionConverter.secTableSections] = listOf(basicDocSection(accNo, type, attributeDocument))
+        return sectionTableDocument
+    }
+
+    private companion object {
+        const val AccNo1 = "accNo1"
```

Comment:
Same comment about constants naming. I think this one also applies for all the others